routers/repo/repo.go

```diff
@@ -170,20 +180,45 @@ func CreatePost(ctx *context.Context, form auth.CreateRepoForm) {
 		return
 	}
 
-	repo, err := repo_service.CreateRepository(ctx.User, ctxUser, models.CreateRepoOptions{
-		Name:        form.RepoName,
-		Description: form.Description,
-		Gitignores:  form.Gitignores,
-		IssueLabels: form.IssueLabels,
-		License:     form.License,
-		Readme:      form.Readme,
-		IsPrivate:   form.Private || setting.Repository.ForcePrivate,
-		AutoInit:    form.AutoInit,
-	})
-	if err == nil {
-		log.Trace("Repository created [%d]: %s/%s", repo.ID, ctxUser.Name, repo.Name)
-		ctx.Redirect(setting.AppSubURL + "/" + ctxUser.Name + "/" + repo.Name)
-		return
+	var err error
+	if form.RepoTemplate > 0 {
+		opts := models.GenerateRepoOptions{
```

Comment:
Should this check again the user permissions and if the repo is actually a template? Perhaps I'm being too paranoid.

Reply:
No that's a good catch, it should look for permissions here.

Reply:
This should now do permission checks as well as validate whether repo is a template or not